app/queries/balance_query.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class BalanceQuery
+  def initialize(user)
+    @user = user
+  end
+
+  def pending_balances
+    pending_debts.merge(pending_credits) { |_, debt, credit| debt + credit }
+  end
+
+  private
+
+  attr_reader :user
+
+  def pending_debts
+    @pending_debts ||= Dish
+      .joins(:order)
+      .where(
+        "orders.status = 1 AND " \
+        "orders.user_id != #{user.id} AND " \
+        "dishes.user_id = #{user.id}"
```

Comment:
String interpolation in SQL queries is basically a bad idea. Here you don't pass any user input but still there's an Active Record way to do it: `Entity.where('field = ? AND field = ?', field_1, field_2)`